src/main/java/com/entropyteam/entropay/employees/services/ContractService.java

```diff
@@ -142,6 +150,7 @@ protected Contract toEntity(ContractDto entity) {
         contract.setSeniority(seniority);
         contract.setPaymentsSettlement(entity.paymentSettlement() == null ? Collections.emptySet() :
                 entity.paymentSettlement().stream().map(PaymentSettlement::new).collect(Collectors.toSet()));
+        contract.setEndReason(endReason);
```

Comment:
@JoaFS another posible implementation of the same is this:
```
Optional.ofNullable(entity.endReasonId())
                .flatMap(endReasonRepository::findById)
                .ifPresent(contract::setEndReason);
```

Both are correct, I'm taking use of this opportunity just to show you some functionalities of the language